### ISSUE TITLE
[java] FP about the rule ConstantsInInterface

### ISSUE THREAD

**Issue by vanguard-1024** [pmd/pmd#4071]
**Affects PMD Version:**
6.47.0+
7.0.0

**Rule:ConstantsInInterface**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.47.0/pmd_rules_java_bestpractices.html#constantsininterface>

**Description:**
Hi, we are doing research in testing static analyzer. Our approach found a false positive about the rule ConstantsInInterface, PMD should not detect aonoymous class for this rule, namely it does not need to report a warning in line 3. It seems that there is a related issue at https://github.com/pmd/pmd/issues/3620. Hence, I think this is a false positive. 

Besides, I find the rule name `ConstantsInInterface` is not appropriate because it can detects different fields, not only constants.

**Code Sample demonstrating the issue:**

```java
public interface Foo {
  public Object o = new Object() {
      public static final int CONST1 = 1; // should not report a warning in this line
  };
}
```

**Expected outcome:**

PMD reports a violation at line 3, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*

**Comment by oowekyala** [comment]
The rule reports on the field `o` here, and not on `CONST1`, so there is nothing wrong here. `o` is a constant field and this is what the rule is designed to flag.
